### PR TITLE
Katabank/sergio

### PR DESCRIPTION
**Pull request by sergioale9**

### CODE REVIEW

File: src/main/java/jalau/at18/katas/bankocr/sergio/PathFile.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package jalau.at18.katas.bankocr.sergio;
+
+import java.io.File;
+
+public class PathFile {
+    private File file;
+    private String pathString = new String(
```
Comment (J0c4) [inline src/main/java/jalau/at18/katas/bankocr/sergio/PathFile.java:7]: Is the usage of this path verified by a test? my gues is that could fail. If this code is part of main logic, it must have at least a unit test to verify it

Usually, the best practice is that reference to files are relative paths instread of full paths. Example, instead of this, relative path might be:
`src\\main\\java\\jalau\\at18\\katas\\bankocr\\sergio\\BankAccounts\\BankAccounts.txt`

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -60,7 +60,7 @@
   <!-- See https://checkstyle.org/config_sizes.html -->
   <module name="FileLength"/>
   <module name="LineLength">
-    <property name="max" value="150"/>
+    <property name="max" value="190"/>
```
Comment (J0c4) [inline config/checkstyle/checkstyle.xml:63]: Please revert this change, we should keep limit to 150. I assume this is because of the file path in one of the files, but it can be reduced by using relative path and also, if even with that it goes beyond 150, break the string in multiple lines, for example, something like this:
```
String alongstring = "asfrikeofjweirojgfeiorj" +
                                  "asfrikeofjweirojgfeiorj" +
                                  "asfrikeofjweirojgfeiorj" +
                                  "asfrikeofjweirojgfeiorj" +
                                  "asfrikeofjweirojgfeiorj";
```

File: build.gradle [hunk 3]
```diff
@@ -27,4 +27,9 @@ checkstyleMain {
 }
 
 test.finalizedBy("checkstyleMain")
-test.finalizedBy("jacocoTestReport")
\ No newline at end of file
+test.finalizedBy("jacocoTestReport")
+
+task runBanckocr (dependsOn: 'classes', type: JavaExec) {
```
Comment (J0c4) [inline build.gradle:32]: This file seems to be having a conflict that prevents merge. 

It is not necessary to add this task so maybe the easiest thing you can do is just revert changes in this file. Your code should be mainly tested by unit tests, so this task would be optional.

File: src/main/java/jalau/at18/katas/bankocr/sergio/ConvertOcrToNumber.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package jalau.at18.katas.bankocr.sergio;
+
+public class ConvertOcrToNumber {
```
Comment (J0c4) [inline src/main/java/jalau/at18/katas/bankocr/sergio/ConvertOcrToNumber.java:3]: Pending?

File: src/main/java/jalau/at18/katas/bankocr/sergio/Main.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package jalau.at18.katas.bankocr.sergio;
+
+import java.io.FileNotFoundException;
+
+public class Main {
+    public static void main(String[] args) throws FileNotFoundException {
+        Numbers numbers = Numbers.EIGHT;
+        System.out.println(numbers.getNumberValue());
```
Comment (J0c4) [inline src/main/java/jalau/at18/katas/bankocr/sergio/Main.java:8]: Looks like this is something that you could do in a unit test

File: src/main/java/jalau/at18/katas/bankocr/sergio/OpticCharacterReader.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
+package jalau.at18.katas.bankocr.sergio;
+
+import java.util.Arrays;
+
+public class OpticCharacterReader {
+    private Numbers numbers;
+
+    public int numberOne(char[][] number) {
+        char[][] numberOne = {{' ', ' ', ' ' }, {' ', ' ', '|' }, {' ', ' ', '|' } };
```
Comment (J0c4) [inline src/main/java/jalau/at18/katas/bankocr/sergio/OpticCharacterReader.java:9]: Maybe you can make this char[][] part of the enum, that way, this reader could be a lot simpler by just getting the ocr representation from the enum.

**Review by J0c4 — CHANGES_REQUESTED** [review]
Hi @sergioale9 , the code looks good, but needs some enhancements. Try to adjust it for not breaking 30 lines limit for classes and some methods to do not break 5 lines limit rule. There is some abstraction for file reading and this is good, but needs unit testing.

Besides that, we can merge, but before that, please add changes for the comments I added in checkstyle.xml file and build.gradle. Those are important to fix before to merge

**Comment by J0c4** [comment]
Hi @sergioale9 , any updates for the comments of my previous review? This cannot be merged before those are addressed

**Comment by J0c4** [comment]
Hello @sergioale9 , following up on this one, any updates?

File: src/main/java/jalau/at18/katas/bankocr/sergio/PathFile.java (hunk 1 above)
Comment (sergioale9) [reply src/main/java/jalau/at18/katas/bankocr/sergio/PathFile.java:7]: solved with relative path

File: config/checkstyle/checkstyle.xml (hunk 2 above)
Comment (sergioale9) [reply config/checkstyle/checkstyle.xml]: file fixed, Sorry for modified it

File: build.gradle (hunk 3 above)
Comment (sergioale9) [reply build.gradle]: fixed

File: src/main/java/jalau/at18/katas/bankocr/sergio/ConvertOcrToNumber.java (hunk 4 above)
Comment (sergioale9) [reply src/main/java/jalau/at18/katas/bankocr/sergio/ConvertOcrToNumber.java:3]: i guess I dont need this file anymore

File: src/main/java/jalau/at18/katas/bankocr/sergio/OpticCharacterReader.java (hunk 6 above)
Comment (sergioale9) [reply src/main/java/jalau/at18/katas/bankocr/sergio/OpticCharacterReader.java]: i was modified the enum class and refactor the opticCharacterReader class for read the pattern numbers of this class

**Review by J0c4 — CHANGES_REQUESTED** [review]
Hi @sergioale9 , latest updates look good, but conflicts seems to still pending, please solve it so we can merge this

**Comment by sergioale9** [comment]
Hi Jose, i solved the problem with the Build.gradle file, and I implemented the users 1 story and user 2 story.
Sorry for delay

**Review by J0c4 — APPROVED** [review]
